### PR TITLE
002 view job spec

### PR DESCRIPTION
**Pull request by rorkainos**
Display the specification of a job role from the job roles ui by clicking on a particular job role

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -17,4 +17,30 @@ router.get('/jobroles', async (req, res) => {
     res.render('jobroles', { jobroles: data } ) 
 });
 
+// render the jobSpec Page with id and name passed in the request
+router.get('/jobspec', async (req, res) => {
+    let id = req.query.id;
```
Comment (shaunganley) [inline app/routes.js:22]: this shouldn't be a query param, you should use /jobspec/:id

File: app/routes.js [hunk 2]
```diff
@@ -17,4 +17,30 @@ router.get('/jobroles', async (req, res) => {
...
+    let name = req.query.name;
+
+    try {
+        jobSpecification = await JobService.getJobSpecification(id);
+
+        let jobSpec = {
+            jobSpecification: jobSpecification.job_spec,
+            jobRoleName:name 
+        }
```
Comment (shaunganley) [inline app/routes.js:31]: remove this and pass the object in to the view

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: app/views/jobSpec.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+   <meta charset="utf-8">
+   <meta name="viewport" content="width=device-width, initial-scale=1">
+   <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
+   <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.1/jquery.min.js"></script>
+   <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
+ </head>
+
+{% block content %}
+
+<h2>Job Specification</h2>
```
Comment (shaunganley) [inline app/views/jobSpec.html:14]: can you add the role name to heading so we know which job spec we're looking at?

File: test/unit/service/JobServiceTests.js [hunk 4]
```diff
@@ -42,8 +42,38 @@ describe('JobService', function () {
...
+
+    describe('getJobSpecification', function () {
+    
+      it('should return the job specification', async () => {
+
+        let data = {job_spec:"spec1"};
+        
+        // mocking a good response from the endpoint
+        var mock = new MockAdapter(axios);
+        mock.onGet(JobService.GET_JOB_SPEC + '1').reply(() => {return [200, data]});
```
Comment (shaunganley) [inline test/unit/service/JobServiceTests.js:61]: why is the function needed inside reply, can't you just .reply(200, data) or similar?

File: test/unit/service/JobServiceTests.js [hunk 5]
```diff
@@ -42,8 +42,38 @@ describe('JobService', function () {
...
+       
+        var response = await JobService.getJobSpecification('1');
+        expect(response.job_spec).to.equal('spec1');
+      })
+
+      it('should return error Could not get Job Specification', async () => {        
+        // mocking an error 500 response from backend
+        var mock = new MockAdapter(axios);
+        mock.onGet(JobService.GET_JOB_SPEC + '1').reply(500);
```
Comment (shaunganley) [inline test/unit/service/JobServiceTests.js:70]: create id var rather than hardcoding

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: app/views/jobSpec.html (hunk 3 above)
Comment (rorkainos) [reply app/views/jobSpec.html:14]: I had this in the tab title, just forgot to change good spot

**Review by KacperBober — APPROVED** [review]

**Review by shaunganley — APPROVED** [review]

**Pull request merged**